File: 2024-02-22 embeddings/embed.py
import ollama

def read_file(file_path):
    with open(file_path, 'r') as file:
        return file.read()


def split_into_chunks(text, chunk_size):
    words = text.split()
    chunks = []
    current_chunk = []

    for word in words:
        current_chunk.append(word)
        if len(current_chunk) >= chunk_size:
            chunks.append(' '.join(current_chunk))
            current_chunk = []

    if current_chunk:  
        chunks.append(' '.join(current_chunk))

    return chunks

file_path = '/Users/matt/Downloads/warandpeace.txt'  #
long_string = read_file(file_path)
chunk_size = 500
chunks = split_into_chunks(long_string, chunk_size)

for i, chunk in enumerate(chunks, start=1):
    response = ollama.embeddings(model='nomic-embed-text', prompt=chunk)
    print(i)
    

File: 2024-02-22 embeddings/embed.ts

import { Ollama } from "ollama";

const ollama = new Ollama();

function splitIntoChunks(text: string, chunkSize: number): string[] {
  const words: string[] = text.split(/\s+/);
  const chunks: string[] = [];
  let currentChunk: string[] = [];
  
  for (const word of words) {
    currentChunk.push(word);
    if (currentChunk.length >= chunkSize) {
      chunks.push(currentChunk.join(' '));
      currentChunk = [];
    }
  }
  if (currentChunk.length > 0) {
    chunks.push(currentChunk.join(' '));
  }
  return chunks;
}


const path = "/Users/matt/Downloads/warandpeace.txt";
const file = Bun.file(path);

const chunks = splitIntoChunks(await file.text(), 500);
console.log('done reading')


let i = 0
for await (const c of chunks) {
  i++;
  const b = await ollama.embeddings({
    model: "llama2",
    prompt: c
  })
  console.log(i);
}

